From 617298fc711e9f5611dbfb44622cef4b330c66f0 Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sat, 13 Nov 2021 11:12:14 -0600
Subject: [PATCH 020/124] ASoC: sun4i-spdif: Assert reset when removing the
 device

This completes reversing the process done in the probe function.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 sound/soc/sunxi/sun4i-spdif.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sound/soc/sunxi/sun4i-spdif.c b/sound/soc/sunxi/sun4i-spdif.c
index a10949bf0ca1..ab42df6da8d8 100644
--- a/sound/soc/sunxi/sun4i-spdif.c
+++ b/sound/soc/sunxi/sun4i-spdif.c
@@ -589,10 +589,14 @@ static int sun4i_spdif_probe(struct platform_device *pdev)
 
 static int sun4i_spdif_remove(struct platform_device *pdev)
 {
+	struct sun4i_spdif_dev *host = dev_get_drvdata(&pdev->dev);
+
 	pm_runtime_disable(&pdev->dev);
 	if (!pm_runtime_status_suspended(&pdev->dev))
 		sun4i_spdif_runtime_suspend(&pdev->dev);
 
+	reset_control_assert(host->rst);
+
 	return 0;
 }
 
-- 
2.20.1

